TransportationSplitter.py

```diff
@@ -506,38 +721,41 @@ def split_segment(input_segment):
             for field_to_drop in input_fields_to_drop_in_splits:
                 original_segment_dict.pop(field_to_drop, None)
 
+            segment_length = get_length(input_segment.geometry)
             debug_messages.append(str(input_segment.geometry))
+            debug_messages.append(f"length={segment_length}")
 
-            split_points = [make_split_point(connector, input_segment.geometry) for connector in input_segment.joined_connectors if connector.connector_geometry is not None]
+            split_points = get_connector_split_points(input_segment.joined_connectors, input_segment.geometry, segment_length) if SPLIT_AT_CONNECTORS else []
 
             debug_messages.append("adding lr split points...")
-            lrs = []
+            lrs_set = set()
             for column in lr_columns_for_splitting:
                 debug_messages.append(f"LRs in [{column}]:")
                 if column not in original_segment_dict or original_segment_dict[column] is None:
                     continue
                 lrs_in_column = get_lrs(original_segment_dict[column])
-                for lr in lrs_in_column:
-                    debug_messages.append(str(lr))
-                lrs = lrs + lrs_in_column
+                debug_messages.append(str(lrs_in_column))
+                lrs_set.update(lrs_in_column)
 
-            add_lr_split_points(split_points, lrs, original_segment_dict["id"], input_segment.geometry)
+            lrs = sorted(lrs_set)
+            debug_messages.append(f"Final LRs set sorted: {lrs}")
+            add_lr_split_points(split_points, lrs, original_segment_dict["id"], input_segment.geometry, segment_length)
 
             sorted_split_points = sorted(split_points, key=lambda p: p.lr)
             debug_messages.append("sorted final split points:")
             for p in sorted_split_points:
-                debug_messages.append(", ".join([p.id, str(p.geometry), str(p.lr)]))
+                debug_messages.append(str(p))
 
             if len(sorted_split_points) < 2:
                 raise Exception(f"Unexpected number of split points: {str(len(sorted_split_points))}; (expected at least 2)")
```

Comment:
It looks like I'm getting a bunch of failures here when trying to not split on connectors

Reply:
I think above instead of having split_points be empty, we should do this instead:
```
            if not SPLIT_AT_CONNECTORS:
                # Only get first and last, downstream logic is dependent on that, ignore other connectors
                split_points = [split_points[0], split_points[-1]]
```
I'm testing that now

Reply:
yeah, that's probably simplest workaround, i'm running a test with that too.

but i'm wondering if this should be though out a bit more, for example it is arguably preferable to reuse existing connectors as much as possible if they exists to correspond to property LR values, rather than introducing new "artificial" connectors like we would with this approach for split_at_connectors=false.
so alternative would be to keep splitpoint objects for all connecters, try to snap LR values to them, create new splitpoints for LRs only when they are not within the "snap" threshold, but then only split at the split points that have lrs  values snapped to them for which the user wants to introduce splits. logic is a bit hairy but would reduce the amount of new connectors that it is adding.

Reply:
included the workaround for now, test worked fine